antiplag/tasks.py

```diff
@@ -41,6 +46,18 @@ def process_documents(submission_id):
     submission.status = Submission.SubmissionStatus.PROCESSED
     submission.save()
 
+    # send email when done
+    if submission.email is not None:
```

Comment:
I don't think the validation should be in here, this the task that goes into the celery. We should validate the email right on the submission endpoint.

Reply:
Submission endpoint saves the email (or none if none was provided) but afterwards, the email shall be sent only when the check is finished, how should we transfer information if the email was valid at the time of submission? (I can only think of branching compare_documents_with_email (without_email) but that seems stupid. So I think this validation is OK at this place.

Reply:
You always want to validate request before doing anything. In this case it's too late. After you saved the submission, preprocessed, and then you find out the email is non-valid, it's too late.

You validate the email on the endpoint, if not, return 400. If it's valid, continue in saving the submission with email to the DB.